staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -5475,7 +5475,7 @@ type ReplicationControllerSpec struct {
 	// Label keys and values that must match in order to be controlled by this replication
 	// controller, if empty defaulted to labels on Pod template.
 	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
-	// +optional
+	// +k8s:required
 	// +mapType=atomic
 	Selector map[string]string `json:"selector,omitempty" protobuf:"bytes,2,rep,name=selector"`
```

Comment:
for +required fields, typically they are not omitempty either.  I want to confirm if this is actually something that can be changed (I don't believe it is)